src/main/java/mateusz/michal/chat/Service/RegistrationService.java

```diff
@@ -28,8 +28,28 @@
     @Autowired
     SlugService slugService;
 
-    @Autowired
-    JsonFactory jsonFactory;
+    public ResponseEntity<IJsonResponse> getResponseForUserRegistration(UserDTO userDTO){
+        List<MyError> errors = validateRegistrationRequest(userDTO);
+        if (errors.size() == 0){
+            User user = createUser(userDTO);
+            userRepository.save(user);
+            return ResponseEntity.ok(
+                    JsonResponseFactory.createResponse(ResponseEnum.DATA,null,
+                            new SimpleDataResponse("user has been succesfully registered and saved in database"),
+                            null));
+        } else {
```

Comment:
Nie potrzebujesz tego `else` - w warunku `if` masz `return`, więc `else` tylko wprowadza niepotrzebne zagnieżdżenie i zamieszanie :)

Reply:
Jest potrzebne. 
Jeśli ilość błędów jest równa 0 to tworzymy usera itd. 
W przeciwnym wypadku przeszukujemy błędy i sprawdzamy czy nie ma błędu z statusem 400
Zauważ, że jeśli warunek errors.size() == 0 będzie false to wpadamy do bloku z else i również  generujemy pewien response.

Reply:
Faktycznie nie jest potrzebne :D

Reply:
W takim razie proszę zmień ;)

Reply:
Zapisałem. 
Zmienię podczas refactoringu struktury projektu który będzie po zakończeniu refactoringu wszystkich response.